Relax concatenation test, backout gracefully

bertsky opened the pull request:
Fixes #5 IMO. Also, it seemed necessary to behave better when text annotation was missing.

mikegerber commented:
"Bei aller Liebe", you could have changed a few lines of code, now you added a Makefile, changed the README, changed setup.py including setting a version, added pylint, did another thing, did some other refactoring, added another change and then did the thing this was supposed to do.

How am I supposed to review this?

mikegerber commented:
This is not about the changes not being useful, I just can't/absolutely do not like to review it like this.

mikegerber commented on setup.py:
```diff
@@ -5,8 +5,10 @@
 
 setup(
     name='ocrd_repair_inconsistencies',
+    version='0.1.0',
```
Please at least remove this version.

mikegerber commented:
I am going to merge this anyway this time, but please remove the version number and open another PR if you think that is necessary.

bertsky commented:
> Bei aller Liebe", you could have changed a few lines of code, now you added a Makefile, changed the README, changed setup.py, added pylint, did another thing, did some other refactoring, added another change and then did the thing this was supposed to do.
> 
> How am I supposed to review this?

I don't see the problem. I made several small, well-described commits. You could simply go through the changes and add your review comments as needed. Or you do it on the final result – with the changelog in mind.

Would you rather have me create 8 separate PRs, each dependent on the previous one?

bertsky commented:
> I am going to merge this anyway this time, but please remove the version number and open another PR if you think that is necessary.

An extra PR just for the version number change? But the changes the new version number captures are in this PR! I don't understand. Don't you want to differentiate versions at all when there is a change in behaviour? How do users update?

mikegerber commented:
> An extra PR just for the version number change? But the changes the new version number captures are in this PR! I don't understand. Don't you want to differentiate versions at all when there is a change in behaviour? How do users update?

With all due respect, it's not up to you to decide on the versioning of this processor. If you think it's necessary to have a version (or even a release instead of just having a git master) please do open an issue. Of course you can propose it like you did and then please accept if I don't want it.

mikegerber commented:
> > Bei aller Liebe", you could have changed a few lines of code, now you added a Makefile, changed the README, changed setup.py, added pylint, did another thing, did some other refactoring, added another change and then did the thing this was supposed to do.
> > How am I supposed to review this?
> 
> I don't see the problem. I made several small, well-described commits. You could simply go through the changes and add your review comments as needed. Or you do it on the final result – with the changelog in mind.
> 
> Would you rather have me create 8 separate PRs, each dependent on the previous one?

At least the added Makefile and pylintrc, half the README changes, the config.py change, the setup.py were absolutely independent - even in meaning re "relax concatenation test" - and could have been a separate PR.

What I don't understand is that people do not get this.

bertsky commented:
> At least the added Makefile and pylintrc, half the README changes, the config.py change, the setup.py were absolutely independent - even in meaning re "relax concatenation test" - and could have been a separate PR.

The README changes were related to the code changes – and the added Makefile. The pylint commit was related to the code changes, too.

> What I don't understand is that people do not get this.

I don't get what this (having it all in one PR) costs you. These are all trivial changes. And making a separate branch and PR for each of them _is_ more effort for the person contributing. Plus there is always the benefit of seeing (and discussing) things in context. (I'm not saying one shouldn't separate PRs that address independent functionality in the code or how it is used BTW.)

mikegerber commented:
> I don't get what this (having it all in one PR) costs you. These are all trivial changes.

It's not so much about cost it's about a change that matches its PR description.
